opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/SingleFieldQuery.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.storage.script.filter.lucene.relevance;
+
+import java.util.Map;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.sql.expression.NamedArgumentExpression;
+
+/**
+ * Base class to represent relevance queries that search a single field only.
+ * @param <T> The builder class for the OpenSearch query class.
+ */
+abstract class SingleFieldQuery<T extends QueryBuilder> extends RelevanceQuery<T> {
```

Comment:
We need a unit test class for this

Reply:
See https://github.com/Bit-Quill/opensearch-project-sql/pull/94/commits/a8b0169dd9badb5a34648f09cefe3ea74972c060